sqlite_query.c

```diff
@@ -412,6 +413,98 @@ sqlite_convert_to_pg(Form_pg_attribute att, sqlite3_value * val, AttInMetadata *
 				}
 				break;
 			}
+		case MACADDROID:
+			{
+				switch (sqlite_value_affinity)
+				{
+					/* SQLite MAC address output always normalized to int64.
+					 * In sqlite_data_norm.c there is special additional C function.
+					 */
+					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							sqlite3_int64		 value = sqlite3_value_int64(val);
+							const sqlite3_uint64 max = (1ULL << (6 * CHAR_BIT)) - 1ULL; /* maximal int64 for macaddr 6b*/
+							macaddr*			 retval;
+							if (value > max )
+								ereport(ERROR,
+										 (errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+										  errmsg("int64 for macaddr out of range")));
+							retval = (macaddr *) palloc(sizeof(macaddr));
+							retval->a = (value >> (CHAR_BIT * 5)) & 0xFF;
+							retval->b = (value >> (CHAR_BIT * 4)) & 0xFF;
+							retval->c = (value >> (CHAR_BIT * 3)) & 0xFF;
+							retval->d = (value >> (CHAR_BIT * 2)) & 0xFF;
+							retval->e = (value >> (CHAR_BIT * 1)) & 0xFF;
+							retval->f = (value >> (CHAR_BIT * 0)) & 0xFF;
+							return (struct NullableDatum){MacaddrPGetDatum(retval), false};
+							break;
```

Comment:
This line will never reach.

Reply:
This is also something from `gcc` warning suppress C code infrastructure.

Reply:
Thank you. Confirmed. Please help me to resolve this comment.